lib/src/zenohUTransport.cpp

```diff
@@ -459,16 +467,15 @@ void ZenohUTransport::QueryHandler(const z_query_t *query, void *arg) {
 
     auto uuidStr = UuidSerializer::serializeToString(attributes.id());
 
-    instance->queryMap_[uuidStr] = z_query_clone(query);
-
     if (UMessageType::UMESSAGE_TYPE_REQUEST != attributes.type()) {
        spdlog::error("Wrong message type = {}", static_cast<int>(attributes.type()));
        return;
     }
 
-    std::unique_lock<std::mutex> lock(instance->queryMapMutex_);
-    instance->queryMap_[uuidStr] = z_query_clone(query);
-    lock.unlock();
+    {
+        std::unique_lock<std::mutex> lock(instance->queryMapMutex_);
+        instance->queryMap_[uuidStr] = z_query_clone(query);
```

Comment:
no unlock?

Reply:
Not necessary. It's [automatic when the variable goes out of scope](https://en.cppreference.com/w/cpp/thread/unique_lock/%7Eunique_lock).